File: examples/python/l3-resources/atlas_basic.py
from aws_cdk import Stack
from constructs import Construct

from awscdk_resources_mongodbatlas import ( AdvancedRegionConfig, AdvancedReplicationSpec,
                                            Specs, AccessListDefinition, IpAccessListProps,
                                            ProjectProps, ClusterProps, AtlasBasic )

class CdkTestAppPython2Stack(Stack):

  def __init__(self, scope: Construct, construct_id: str, **kwargs) -> None:
    super().__init__(scope, construct_id, **kwargs)

    org_id_var = self.node.try_get_context('org_id')
    region_var = self.node.try_get_context('region')
    profile_name_var = self.node.try_get_context('profile')
    ip_addr_var = self.node.try_get_context('ip_addr')
    ip_comment_var = self.node.try_get_context('ip_comment')

    region_configs_var = []
    region_configs_var.append(
      AdvancedRegionConfig(analytics_specs=Specs(node_count=1, instance_size="M10", ebs_volume_type="STANDARD"),
                           electable_specs=Specs(node_count=3, instance_size="M10", ebs_volume_type="STANDARD"),
                           priority=7,
                           region_name=region_var))
    replication_specs_var = []
    replication_specs_var.append(AdvancedReplicationSpec(advanced_region_configs = region_configs_var, num_shards = 1))

    access_list_defs_var = []
    access_list_defs_var.append(AccessListDefinition(ip_address = ip_addr_var, comment = ip_comment_var))

    atlas_basic_l3 = AtlasBasic(self, "AtlasBasic-py-l3",
                                      cluster_props=ClusterProps(replication_specs=replication_specs_var),
                                      project_props=ProjectProps(org_id=org_id_var),
                                      ip_access_list_props=IpAccessListProps(access_list = access_list_defs_var),
                                      profile=profile_name_var)

File: examples/java/l3-resources/AtlasBasic.java
package com.myorg;

import java.util.List;
import org.jetbrains.annotations.NotNull;
import software.constructs.Construct;
import software.amazon.awscdk.Stack;
import software.amazon.awscdk.StackProps;
import org.mongodb.awscdk.resources.mongodbatlas.*;
import software.constructs.Node;


public class CdkJavaStack extends Stack {
    final static String ORG_ID = "ORG_ID";
    final static String CLUSTER_NAME = "CLUSTER_NAME";
    final static String ACCESS_LIST_IP = "ACCESS_LIST_IP";
    final static String PROFILE = "PROFILE";
    final static String PROJECT_NAME = "PROJECT_NAME";


    public CdkJavaStack(final Construct scope, final String id, final StackProps props) {
        super(scope, id, props);

        final String clusterName = getContextProp(this.getNode(), CLUSTER_NAME);
        final String orgId = getContextProp(this.getNode(), ORG_ID);
        final String profile = getContextProp(this.getNode(), PROFILE);
        final String ip = getContextProp(this.getNode(), ACCESS_LIST_IP);
        final String projectName = getContextProp(this.getNode(), PROJECT_NAME);

        final AtlasBasic atlasBasic = AtlasBasic.Builder.create(this, "atlasBasic")
            .profile(profile)
            .ipAccessListProps(
                IpAccessListProps.builder().accessList(
                    List.of(AccessListDefinition.builder()
                        .ipAddress(ip).comment("CDK java").build())
                ).build())
            .projectProps(ProjectProps.builder()
                .orgId(orgId)
                .name(projectName)
                .build())
            .clusterProps(ClusterProps.builder()
                .name(clusterName)
                .replicationSpecs(List.of(
                    AdvancedReplicationSpec.builder()
                        .numShards(1)
                        .advancedRegionConfigs(
                            List.of(
                                AdvancedRegionConfig.builder()
                                    .regionName("US_EAST_1")
                                    .priority(7)
                                    .analyticsSpecs(
                                        Specs.builder()
                                            .ebsVolumeType("STANDARD")
                                            .instanceSize("M10")
                                            .nodeCount(1)
                                            .build())
                                    .electableSpecs(
                                        Specs.builder()
                                            .ebsVolumeType("STANDARD")
                                            .instanceSize("M10")
                                            .nodeCount(3)
                                            .build())
                                    .build()
                            )
                        )
                        .build()
                ))
                .build())
            .build();
    }

    public String getContextProp(final Node pNode, final String pPropName){
        @NotNull final Object obj = pNode.tryGetContext(pPropName);
        if (obj.toString().isEmpty()) {
            return "";
        }
        return obj.toString();
    }
}
